L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -0,0 +1,462 @@
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > 113.) && (global_phi <= 135.)) {
+	    local_phi = global_phi-75.;
+	    EMTFsector1 = 2;
+	    EMTFsector2 = 3;
+	  }
+	  //sector 3
+	  //less than 180
+	  if ((global_phi > 135.) && (global_phi <= 136.3)) {
+	    local_phi = global_phi-75.;
+	    EMTFsector1 = 2;
+	    EMTFsector2 = 3;
+	  }
+	  if ((global_phi > 136.3) && (global_phi <= 173.)) {
+	    local_phi = global_phi-135.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > 173.) && (global_phi <= 180.)) {
+	    local_phi = global_phi-135.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  //Greater than -180
+	  if ((global_phi < -165.) && (global_phi >= -180.)) {
+	    local_phi = global_phi+225.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  //Fourth sector
+	  if ((global_phi > -165.) && (global_phi <= -163.7)) {
+	    local_phi = global_phi+225.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  if ((global_phi > -163.7) && (global_phi <= -127.)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -127.) && (global_phi <= -105.)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 5;
+	  }
+	  //fifth sector
+	  if ((global_phi > -105.) && (global_phi <= -103.7)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 5;
+	  }
+	  if ((global_phi > -103.7) && (global_phi <= -67.)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -67.) && (global_phi <= -45.)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 6;
+	  } 
+	  //sixth sector
+	  if ((global_phi > -45.) && (global_phi <= -43.7)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 6;
+	  } 
+	  if ((global_phi > -43.7) && (global_phi <= -7.)) {
+	    local_phi = global_phi+45.;
+	    EMTFsector1 = 6;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -7.) && (global_phi <= 15.)) {
+	    local_phi = global_phi+45.;
+	    EMTFsector1 = 6;
+	    EMTFsector2 = 1;
+	  } 
+	  
+	  int int_phi = int((local_phi + 22.0 )*15. + .5); 
+	  
+	  double EMTFLink1 = 0.;
+	  double EMTFLink2 = 0.;
+          double lb = 0.;
+	  double halfchannel = 0.;
+	  
+	  //Invalid hit
+	  if (isValid == 0) int_phi = 2047;	 
+	  //Right integers range
+	  assert(0 <= int_phi && int_phi < 1250);
+	  assert(0 <= int_theta && int_theta < 32);
+	  
+	  l1t::CPPFDigi* MainVariables1 = new l1t::CPPFDigi(rpcId, Bx , int_phi, int_theta, isValid, lb, halfchannel, EMTFsector1, EMTFLink1, firststrip, clustersize, global_phi, global_theta);
+	  l1t::CPPFDigi* MainVariables2 = new l1t::CPPFDigi(rpcId, Bx , int_phi, int_theta, isValid, lb, halfchannel, EMTFsector2, EMTFLink2, firststrip, clustersize, global_phi, global_theta);
+	  if(int_theta == 31) continue;
+          else if ((EMTFsector1 > 0) && (EMTFsector2 == 0)){
```

Comment:
No need for an `else if` here since there is the `continue` in the `if` before. It is a bit confusing to compare different variables in the if and the else ifs.